airbyte-ci/connectors/metadata_service/lib/metadata_service/registry.py

```diff
@@ -0,0 +1,373 @@
+
+    Args:
+        registry_entry (PolymorphicRegistryEntry): The registry entry.
+
+    Returns:
+        ConnectorTypes: The connector type.
+    """
+    if hasattr(registry_entry, ConnectorTypePrimaryKey.SOURCE):
+        return ConnectorTypes.SOURCE
+    elif hasattr(registry_entry, ConnectorTypePrimaryKey.DESTINATION):
+        return ConnectorTypes.DESTINATION
+    else:
+        raise ValueError("Registry entry is not a source or destination")
+
+
+@sentry_sdk.trace
+def _get_latest_registry_entries(bucket: storage.Bucket, registry_type: str) -> list[PolymorphicRegistryEntry]:
+    """Get the latest registry entries from the GCS bucket.
+
+    Args:
+        bucket (storage.Bucket): The GCS bucket.
+        registry_type (str): The registry type.
+
+    Returns:
+        list[PolymorphicRegistryEntry]: The latest registry entries.
+    """
+    registry_type_file_name = f"{registry_type}.json"
+
+    try:
+        logger.info(f"Listing blobs in the latest folder: {METADATA_FOLDER}/**/latest/{registry_type_file_name}")
+        blobs = bucket.list_blobs(match_glob=f"{METADATA_FOLDER}/**/latest/{registry_type_file_name}")
+    except Exception as e:
+        logger.error(f"Error listing blobs in the latest folder: {str(e)}")
+        return []
+
+    latest_registry_entries = []
+    for blob in blobs:
+        logger.info(f"Reading blob: {blob.name}")
+        registry_dict = json.loads(safe_read_gcs_file(blob))
+        try:
+            if registry_dict.get(ConnectorTypePrimaryKey.SOURCE.value):
+                registry_model = ConnectorRegistrySourceDefinition.parse_obj(registry_dict)
+            elif registry_dict.get(ConnectorTypePrimaryKey.DESTINATION.value):
+                registry_model = ConnectorRegistryDestinationDefinition.parse_obj(registry_dict)
+            else:
+                logger.warning(f"Failed to parse registry model for {blob.name}. Skipping.")
+                continue
+        except Exception as e:
+            logger.error(f"Error parsing registry model for {blob.name}: {str(e)}")
+            continue
+        latest_registry_entries.append(registry_model)
+    return latest_registry_entries
+
+
+@sentry_sdk.trace
+def _get_release_candidate_registry_entries(bucket: storage.Bucket, registry_type: str) -> list[PolymorphicRegistryEntry]:
+    """Get the release candidate registry entries from the GCS bucket.
+
+    Args:
+        bucket (storage.Bucket): The GCS bucket.
+        registry_type (str): The registry type.
+
+    Returns:
+        list[PolymorphicRegistryEntry]: The release candidate registry entries.
+    """
+    blobs = bucket.list_blobs(match_glob=f"{METADATA_FOLDER}/**/release_candidate/{registry_type}.json")
+    release_candidate_registry_entries = []
+    for blob in blobs:
+        logger.info(f"Reading blob: {blob.name}")
+        registry_dict = json.loads(safe_read_gcs_file(blob))
+        try:
+            if "source-" in blob.name:
```

Comment:
```suggestion
            if "/source-" in blob.name:
```
... on the very low chance that someone creates `destination-source-foo`